frontend/public/components/factory/table.tsx

```diff
@@ -0,0 +1,600 @@
+const sorts = {
+  alertStateOrder,
+  daemonsetNumScheduled: daemonset => _.toInteger(_.get(daemonset, 'status.currentNumberScheduled')),
+  dataSize: resource => _.size(_.get(resource, 'data')),
+  ingressValidHosts,
+  serviceCatalogStatus,
+  jobCompletions: job => getJobTypeAndCompletions(job).completions,
+  jobType: job => getJobTypeAndCompletions(job).type,
+  nodeReadiness: node => {
+    let readiness = _.get(node, 'status.conditions');
+    readiness = _.find(readiness, {type: 'Ready'});
+    return _.get(readiness, 'status');
+  },
+  numReplicas: resource => _.toInteger(_.get(resource, 'status.replicas')),
+  planExternalName,
+  podPhase,
+  podReadiness,
+  serviceClassDisplayName,
+  silenceStateOrder,
+  string: val => JSON.stringify(val),
+  getClusterOperatorStatus,
+  getClusterOperatorVersion,
+  getTemplateInstanceStatus,
+  nodeRoles: (node: K8sResourceKind): string => {
+    const roles = getNodeRoles(node);
+    return roles.sort().join(', ');
+  },
+};
+
+const stateToProps = ({UI}, {data = [], defaultSortField = 'metadata.name', defaultSortFunc = undefined, filters = {}, loaded = false, reduxID = null, reduxIDs = null, staticFilters = [{}] }) => {
+  const allFilters = staticFilters ? Object.assign({}, filters, ...staticFilters) : filters;
+  let newData = getFilteredRows(allFilters, data);
+
+  const listId = reduxIDs ? reduxIDs.join(',') : reduxID;
+  // Only default to 'metadata.name' if no `defaultSortFunc`
+  const currentSortField = UI.getIn(['listSorts', listId, 'field'], defaultSortFunc ? undefined : defaultSortField);
+  const currentSortFunc = UI.getIn(['listSorts', listId, 'func'], defaultSortFunc);
+  const currentSortOrder = UI.getIn(['listSorts', listId, 'orderBy'], SortByDirection.asc);
+
+  if (loaded) {
+    let sortBy: string | Function = 'metadata.name';
+    if (currentSortField) {
+      // Sort resources by one of their fields as a string
+      sortBy = resource => sorts.string(_.get(resource, currentSortField, ''));
+    } else if (currentSortFunc && sorts[currentSortFunc]) {
+      // Sort resources by a function in the 'sorts' object
+      sortBy = sorts[currentSortFunc];
+    }
+
+    // Always set the secondary sort criteria to ascending by name
+    newData = _.orderBy(newData, [sortBy, 'metadata.name'], [currentSortOrder, SortByDirection.asc]);
+  }
+
+  return {
+    currentSortField,
+    currentSortFunc,
+    currentSortOrder,
+    data: newData,
+    listId,
+  };
+};
+
+// Common table row/columns helper SFCs for implementing accessible data grid
+export const Vr: React.SFC<VrProps> = ({id, index, trKey, style, className, ...props}) => {
+  return (
+    <tr {...props} data-id={id} data-index={index} data-key={trKey} style={style} className={className} role="row" />
+  );
+};
+Vr.displayName = 'Vr';
+export type VrProps = {
+  id: any;
+  index: number;
+  trKey: string;
+  style: object;
+  className?: string;
+}
+
+export const Vd: React.SFC<VdProps> = ({className, ...props}) => {
```

Comment:
`VD`. I'm hesitant to use this acronym since it has other meanings. `TableRow` and `TableColumn` might be better even if longer.

Reply:
renamed to `TableRow` and `TableData`